comit/src/network/protocols/orderbook.rs

```diff
@@ -160,39 +163,43 @@ impl Orderbook {
         self.orders.get(id).cloned()
     }
 
-    pub fn get_trading_pairs(&self) -> Vec<TradingPairTopic> {
-        self.trading_pairs.iter().cloned().collect()
-    }
-
-    /// Add a trading pair, called when we receive a topic via gossipsub.
-    pub fn add_trading_pair(&mut self, topic: &TradingPairTopic) {
-        self.trading_pairs.insert(topic.clone());
+    fn is_subscribed(&self, topic: &str) -> bool {
+        self.subs.contains(topic)
     }
 
     /// Subscribe to a trading pair topic from peer.
-    pub fn subscribe(&mut self, peer: PeerId, trading_pair: TradingPair) -> anyhow::Result<()> {
-        let topic = TradingPairTopic::new(peer, trading_pair);
-        self.trading_pairs.insert(topic.clone());
-        self.gossipsub.subscribe(topic.to_topic());
-        Ok(())
+    /// Returns true if the subscription worked. Returns false if we were
+    /// already subscribed.
+    pub fn subscribe(&mut self, topic: String) -> bool {
```

Comment:
Why do we have `String`-based APIs in a strongly typed language? :)

Why are topics even exposed on the public API? How does this go together with making the use of gossipsub and how we design the topics an implementation detail?

Reply:
You scratched a good itch here. We actually do not need to handle subscriptions at all, gossipsub does this for us. All this code is now removed.